tests/ert/unit_tests/gui/tools/plot/test_plot_api.py

```diff
@@ -283,3 +283,41 @@ def test_plot_api_handles_empty_gen_kw(tmp_path, monkeypatch):
             enkf._storage.close()
         enkf._storage = None
         gc.collect()
+
+
+def test_plot_api_handles_non_existant_gen_kw(tmp_path, monkeypatch):
+    with open_storage(tmp_path / "storage", mode="w") as storage:
+        monkeypatch.setenv("ERT_STORAGE_NO_TOKEN", "yup")
+        monkeypatch.setenv("ERT_STORAGE_ENS_PATH", storage.path)
+        api = PlotApi()
+        experiment = storage.create_experiment(
+            parameters=[
+                GenKwConfig(
+                    name="gen_kw",
+                    forward_init=False,
+                    update=False,
+                    template_file=None,
+                    output_file=None,
+                    transform_function_definitions=[],
+                ),
+            ],
+            responses=[],
+            observations={},
+        )
+        ensemble = storage.create_ensemble(experiment.id, ensemble_size=10)
+        ensemble.save_parameters(
+            "gen_kw",
+            1,
+            xr.Dataset(
+                {
+                    "values": ("names", [1.0]),
+                    "transformed_values": ("names", [1.0]),
+                    "names": ["key"],
+                }
+            ),
+        )
+        assert api.data_for_key(str(ensemble.id), "gen_kw").empty
```

Comment:
Can you use some other key than ensemble.id? eg. `key_not_present` as `ensemble.id` makes you think :smile:

Reply:
`ensemble.id` has to exist, and so has `gen_kw`, the bug is triggered by asking for `gen_kw` rather than `gen_kw:key`.

Reply:
right, then let's do `gen_kw:key_not_present`

Reply:
```python
assert api.data_for_key(str(ensemble.id), "gen_kw").empty
assert api.data_for_key(str(ensemble.id), "gen_kw:key_not_present").empty
```